### DISCUSSION TITLE
WalletConnect 2.0

### DISCUSSION THREAD

**Discussion by shan-c** [wevm/wagmi#2582] [Q&A]
Is there any open source dapp that implemented wallet connect 2.0 with wagmi ?

**Comment by kristiegiles** [comment]
These are the two remaining issues preventing me from switching to the wagmi WalletConnect v2 connector in prod:

- MetaMask isn't available as a connection option yet (but is now available for v2 in other libraries)
- It [fails to connect on iOS 15 devices](https://github.com/wagmi-dev/wagmi/issues/2590)

I also had a problem with the WalletConnect searchbox not working, but I tracked this down to a conflict with react-focus-lock and was able to fix it using their whiteList option to restrict focus lock to my Next.js app root.

**Reply by jxom** [reply]
These issues seem to stem from the WalletConnect `ethereum-provider` package. Would you mind opening a discussion there? 

https://github.com/WalletConnect/walletconnect-monorepo

**Reply by kristiegiles** [reply]
Thanks! ✅ [https://github.com/orgs/WalletConnect/discussions/2745](https://github.com/orgs/WalletConnect/discussions/2745)

**Reply by kristiegiles** [reply]
Both these issues are fixed now in v1.3.1!

**Comment by quagliero** [comment]
I'm getting the `"wallet_addEthereumChain" not implemented` error on our wagmi apps when trying to connect via Safe (walletconnect).

This issue was reported: https://github.com/safe-global/safe-react-apps/issues/709

And then resolved: https://github.com/safe-global/safe-react-apps/pull/706/

Just wondering how this now works its way back into wagmi. And specifically if it can be fixed on apps that run 0.12.x of wagmi..

**Reply by t0rbik** [reply]
Having same issue on viem 1.2.6 and wagmi 1.3.3 on fantom with fantom safe wallet connect app. Kind of lost at this point

**Reply by diegoalzate** [reply]
I am trying to connect to dapp with wallet connect v2 and wagmi and then deploy a safe with `safe-protocol-kit`. I can connect pretty fine, when using the ethers adapter proposed in the [docs](https://wagmi.sh/react/ethers-adapters), I'm not able to create a [SafeFactory](https://github.com/safe-global/safe-core-sdk/tree/main/packages/protocol-kit)

**Reply by tmm** [reply]
@quagliero you can install from the `legacy-v0` tag. We are making sure WalletConnect v2 bug fixes make it there.

**Reply by quagliero** [reply]
Thanks @tmm 💙

**Comment by Illyism** [comment]
Is there a fix for wagmi v0.1x?

**Reply by quagliero** [reply]
I upgraded to 0.12.x (along with rainbowkit 0.12.x) and it worked.

Additional note that if using a nextjs app you’ll also need to be running >=0.16 node version as it kept failing to open the modal when node 0.14 runtime. This may be a total red herring but figure I’d add it here incase it helps someone.
